### PR TITLE
feat(radiocomponent): create radio component

### PR DESCRIPTION
**Pull request by fakela**
What does this PR do
This PR creates radio components

**Description of Task to be completed**
Create a reusable radio component according to the style guide.

**How should this be manually tested**
By opening up storybook and viewing the radio component.
Storybook can be run with the command npm run storybook

**What are the relevant issues this PR belongs to**
#19

Any background context you want to add
N/A

### CODE REVIEW

File: src/components/RadioButton.js/RadioButton.stories.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';;
+import  RadioButton from '.';
+import styles from './style.module.css';
```
Comment (ashinzekene) [inline src/components/RadioButton.js/RadioButton.stories.js]: This file does not exist

Comment (ashinzekene) [reply src/components/RadioButton.js/RadioButton.stories.js]: You can also check how to use css modules [here](https://github.com/css-modules/css-modules/blob/master/README.md)

File: src/components/RadioButton.js/index.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React, { Children } from 'react';
+import classNames from 'classnames';
+import styles from './style.module.css';
```
Comment (ashinzekene) [inline src/components/RadioButton.js/index.js]: You can also check how to use css modules [here](https://github.com/css-modules/css-modules/blob/master/README.md)

File: src/components/RadioButton.js/index.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import React, { Children } from 'react';
+import classNames from 'classnames';
```
Comment (ashinzekene) [inline src/components/RadioButton.js/index.js]: Not used too

**Comment by ashinzekene** [comment]
Looks good but from the styleguide, There are three radio button types

1. Primary (dark-blue)
2. Secondary (light-bluish)
3. Disabled


Currently, your `disabled` is the `secondary`

File: package.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
   "dependencies": {
     "@storybook/addon-storysource": "^5.2.8",
     "classnames": "2.2.6",
-    "gatsby": "2.17.15",
+    "gatsby": "^2.18.6",
```
Comment (ashinzekene) [inline package.json]: @BolajiAyodeji is this fine?

Comment (fakela) [reply package.json]: yeah when i did npm install that changed

**Comment by fakela** [comment]
radiobuttons are supposed to be 
1) the normal
2)checked
3)disabled and 
4)disabled checked

and from the styleguide i saw something really to faint for my eyes to see more reason i didnt eveb bother looking at it and besides what category do they fall in

File: package.json (hunk 4 above)
Comment (BolajiAyodeji) [reply package.json]: I don't think `npm install` should upgrade gatsby for you. Are you certain that this got upgraded when you ran `npm install`?

**Comment by fakela** [comment]
Yeah it did cause I never opened the package.json file

**Comment by fakela** [comment]
> > radiobuttons are supposed to be
> > 
> > 1. the default
> >    2)checked
> >    3)disabled and
> >    4)disabled checked
> > 
> > and from the styleguide i saw something really to faint for my eyes to see more reason i didnt even bother looking at it and besides what category do they fall in
> > @ashinzekene @BolajiAyodeji
> 
> The following states pseudo-classes can be applied to a radio element:
> 
> * default (:unchecked)
> * :checked
> * :enabled
> * :disabled
> 
> I don't think there should be a `disabled checked` as disabled means the element is not interactive and `checked` says otherwise.
> 
> Can you check this, please?

Alright I will check it

File: src/components/RadioButton.js/RadioButton.stories.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';;
+import  RadioButton from '.';
```
[the remark this answers is not in the document]
Comment (fakela) [reply src/components/RadioButton.js/RadioButton.stories.js]: > There's some extra space here.

My bad I didn't notice that

**Comment by BolajiAyodeji** [comment]
Is this ready for a MR @fakela ?

**Comment by fakela** [comment]
> Is this ready for a MR @fakela ?

i guess so

File: src/components/RadioButton.js/index.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+                type="radio"
+                name={name}
+                onChange={onChange}
+                onBlur={onBlur}
+                value={value}
+                checked={checked}
+                  {...props}
+            />
+            {children}
+        <label className={_fieldClass}style={{...style}} />
```
Comment (ashinzekene) [inline src/components/RadioButton.js/index.js]: This fails for [accessibility](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-associated-control.md), the label should be wrapped around the input

File: src/components/RadioButton.js/RadioButton.stories.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+export default {
+  title: 'RadioButton',
+  component: 'RadioButton'
+};
+
+
+const onChange = () => console.log('Value has changed');
+
+
+   export const defaultRadioButton = () =>  <RadioButton  />
```
Comment (ashinzekene) [inline src/components/RadioButton.js/RadioButton.stories.js]: Nice work on this but some of code doesn't seem well formatted, you can run `npm run fix:lint` to fix most of them

File: src/components/RadioButton.js/index.js (hunk 6 above)
Comment (fakela) [reply src/components/RadioButton.js/index.js]: The label cannot be wrapped inside the input it will throw an error

Comment (ashinzekene) [reply src/components/RadioButton.js/index.js]: image

image

Comment (fakela) [reply src/components/RadioButton.js/index.js]: Screenshot 2019-12-09 at 8 30 47 AM

this should be the right way

**Comment by fakela** [comment]
any other changes you want me to make

File: src/components/RadioButton.js/index.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+                  {...props}
+            />
+            {children}
+    </label>
+        </div>
+        </div>
+        
+    );
+};
+};
```
Comment (ashinzekene) [inline src/components/RadioButton.js/index.js]: I don't think this should be there

File: src/components/RadioButton.js/index.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+
+function RadioButton({value, onChange,label,style, onBlur, name, children, className,checked, ...props}) {
+    const _className = classNames(styles.oc_radiobutton_container,styles.oc_field,className);
+    const _fieldClass = classNames(styles.oc_field_label,styles.oc_label, className)
+    const _wrapper = classNames(styles.oc_wrapper, className)
+    const _item = classNames(styles.oc_item, className)
+
+   return (
+        <div className={_wrapper}>
```
Comment (ashinzekene) [inline src/components/RadioButton.js/index.js]: Little formatting here. And I don't think you need two wrapper divs. Something like this should be fine. You would need to update the styling to merge the two divs though

```jsx
<div className={_wrapper}>
    <label className={_fieldClass}style={{...style}}>
        <input 
            className={_className}
            type="radio"
            name={name}
            onChange={onChange}
            onBlur={onBlur}
            value={value}
            checked={checked}
            {...props}
        />
        {children}
    </label>
</div>
```

File: src/components/RadioButton.js/index.js (hunk 6 above)
Comment (ashinzekene) [reply src/components/RadioButton.js/index.js]: Yeah

**Comment by ashinzekene** [comment]
Apart from the issues listed above and the following 
- There is no real example of how to use multiple radios
- The radios are supposed to [controlled](https://reactjs.org/docs/forms.html#controlled-components) instead of [uncontrolled](https://reactjs.org/docs/uncontrolled-components.html)

which are really not so important. It looks good

File: src/components/RadioButton.js/index.js (hunk 8 above)
Comment (fakela) [reply src/components/RadioButton.js/index.js]: ohhh....that was an extra will take that off

**Comment by ashinzekene** [comment]
Also the folder should be named `RadioButton` instead of `RadioButton.js`. Sorry I didn't find this earlier

**Comment by fakela** [comment]
> Also the folder should be named `RadioButton` instead of `RadioButton.js`. Sorry I didn't find this earlier

any other edits aside that

**Comment by ashinzekene** [comment]
I see you removed other div in the JSX but you didn't remove the style [here](https://github.com/oscommunityafrica/osf-website/pull/45/files#diff-ba73eefa18de9461753cfce619bdef75)

**Comment by fakela** [comment]
I guess this is ready for a MR

File: src/components/RadioButton/index.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+
+function RadioButton({value, onChange,label,style, onBlur, name, children, className,checked, ...props}) {
+    const _className = classNames(styles.oc_radiobutton_container,styles.oc_field,className);
+    const _fieldClass = classNames(styles.oc_field_label, className)
+    const _wrapper = classNames(styles.oc_wrapper,styles.oc_item, className)
+  
+    return (
+        <div className={_wrapper}>
+            <label className={_fieldClass}style={{...style}}>
```
Comment (ashinzekene) [inline src/components/RadioButton/index.js]: You forgot a space here 😄 
```suggestion
            <label className={_fieldClass} style={{...style}}>
```

**Comment by ashinzekene** [comment]
> I see you removed other div in the JSX but you didn't remove the style [here](https://github.com/oscommunityafrica/osf-website/pull/45/files#diff-ba73eefa18de9461753cfce619bdef75)

Yeah looks good, but you forgot to remove `oc_item` in this [file](https://github.com/oscommunityafrica/osf-website/pull/45/files#diff-ba73eefa18de9461753cfce619bdef75) since it's no longer used, and then check for inconsistent spacings

**Comment by fakela** [comment]
Lol...will get on this later today

**Comment by fakela** [comment]
Thank you @ashinzekene @BolajiAyodeji  for the great review 👍

File: src/components/RadioButton/index.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+                onChange={onChange}
+                onBlur={onBlur}
+                value={value}
+                checked={checked}
+                {...props}
+            />
+            {children}
+    </label>
+        </div>
+
```
Comment (ashinzekene) [inline src/components/RadioButton/index.js]:
```jsx
        <div className={_wrapper}>
            <label className={_fieldClass} style={{...style}}>
                <input 
                    className={_className}
                    type="radio"
                    name={name}
                    onChange={onChange}
                    onBlur={onBlur}
                    value={value}
                    checked={checked}
                    {...props}
                />
                {children}
            </label>
        </div>
```

**Comment by ashinzekene** [comment]
> > Also the folder should be named `RadioButton` instead of `RadioButton.js`. Sorry I didn't find this earlier
> 
> any other edits aside that @ashinzekene can you make this change yourself
> 
> > I see you removed other div in the JSX but you didn't remove the style [here](https://github.com/oscommunityafrica/osf-website/pull/45/files#diff-ba73eefa18de9461753cfce619bdef75)
> 
> Screenshot 2019-12-09 at 12 03 12 PM
> 
> the style is declared here and its needed for actual placement of the div

Oh, I'm just seeing this now

**Comment by ashinzekene** [comment]
You're welcome @BolajiAyodeji and @fakela 

After this, it's fine to merge

File: src/components/RadioButton/index.js (hunk 11 above)
Comment (fakela) [reply src/components/RadioButton/index.js]: lol I don't have the linting plugins and extensions on my IDE so I can't really fix all this spaces and lines 
maybe you could do that

**Comment by ashinzekene** [comment]
Nice work @fakela !


@BolajiAyodeji this PR is ready now

**Comment by fakela** [comment]
> Nice work @fakela !
> 
> @BolajiAyodeji this PR is ready now

pheeew....finally....thanks @ashinzekene

File: src/components/RadioButton/index.js (hunk 11 above)
Comment (BolajiAyodeji) [reply src/components/RadioButton/index.js]: Can you set it up instead? You will always need it.

Comment (fakela) [reply src/components/RadioButton/index.js]: sure i will...thanks

Comment (ashinzekene) [reply src/components/RadioButton/index.js]: Actually the current ESLint rules does not cover these. I'd create a PR to enforce and correct these issues

**Comment by BolajiAyodeji** [comment]
@allcontributors please add @fakela for code

**Pull request merged**